src/libraries/System.Text.Json/src/System/Text/Json/Serialization/JsonSerializerOptions.cs

```diff
@@ -198,27 +175,32 @@ public IJsonTypeInfoResolver TypeInfoResolver
             [RequiresDynamicCode(JsonSerializer.SerializationRequiresDynamicCodeMessage)]
             get
             {
-                if (_typeInfoResolver != null)
-                {
-                    return _typeInfoResolver;
-                }
-
-                EnsureInitializedForReflectionSerializer();
-                return _typeInfoResolver;
+                return _typeInfoResolver ?? DefaultJsonTypeInfoResolver.DefaultInstance;
```

Comment:
`??=`?

Reply:
do I understand correctly `_typeInfoResolver` is null unless it's assigned explicitly?

Reply:
Yeah, we don't want to force configuration just because we're calling a get method. I'm still thinking about potential making the property nullable (and only populate it once options is locked) so that we remove the dynamic code annotation.

Reply:
Lots of open questions TBD here.